### ISSUE TITLE
most module tab went missing in skipper mode

### ISSUE THREAD

**Issue by exzine** [spring-cloud/spring-cloud-dataflow-ui#973]
most module tabs went missing except app/about. other modules are still accessible by explicitly type in the url. the information on about module will keep loading forever.

I am using latest scdf for cf on a pcfdev vm.

**Comment by sabbyanandan** [comment]
Hi, @exzine. The report doesn't include version numbers, so I'm going to assume you're on v1.7 and running the CF-server on PCFDev. All the symptoms described here relates to the CF-server not having a successful connection with Skipper. Please follow the [getting-started instruction](http://docs.spring.io/spring-cloud-dataflow-server-cloudfoundry/docs/1.7.0.RELEASE/reference/htmlsingle/#getting-started-cloudfoundry) in entirety. 

In other words, if you're configuring SCDF to be on skipper-mode, you also have to push the Skipper-server and provide it's API URI/route as a configuration property to SCDF's CF-server.

Hope this helps.

**Comment by exzine** [comment]
hi Sabby,

thanks for your reply. I do push the skipper server 1.1.0 to PCFDEV, no
error reported. I can see a skipper app runing in apps manager, but can't
connect through shell.

server-unknown:>skipper config --uri https://mlp-skipper.dev.cfdev.sh/api
I/O error on GET request for "https://mlp-skipper.dev.cfdev.sh/api/about":
sun.security.validator.ValidatorException: PKIX path building failed:
sun.security.provider.certpath.SunCertPathBuilderException: unable to find
valid certification path to requested target; nested exception is
javax.net.ssl.SSLHandshakeException:
sun.security.validator.ValidatorException: PKIX path building failed:
sun.security.provider.certpath.SunCertPathBuilderException: unable to find
valid certification path to requested target
Details of the error have been omitted. You can use the stacktrace command
to print the full stacktrace.

this might related to the anomaly in SCDF (v1.7) skipper mode as you
suggested.

**Comment by exzine** [comment]
Thanks Sabby,

I replaced the https connection with a normal http from SPRING_CLOUD_SKIPPER_CLIENT_SERVER_URI: http://mlp-skipper.dev.cfdev.sh/api , and all things worked out.

I think it might be good to have a environment toggle in the SCDF manifest similar as SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SKIP_SSL_VALIDATION to handle it.

**Comment by sabbyanandan** [comment]
Thanks for the confirmation. Setting `SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SKIP_SSL_VALIDATION` to "true" should have automatically taken care of it, but it didn't in your case. Feel free to open a follow-up issue in [CF-server](https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry) - we will review it on PCFDev.
